src/extensions/birthday_event.py

```diff
@@ -0,0 +1,292 @@
+"""Birthday Events: Creates scheduled guild events for
+member birthdays and un-birthdays.
+
+A server member with `MANAGE_EVENTS` permissions may use
+\opt-in-server-to-birthday-events to opt in a server
+for birthday events and choose a channel for the events, or
+use \opt_out_server_from_birthday_events to remove the server
+from participating in birthday events.
+
+The associated functions, `opt_out_server_from_birthday_events`
+and `opt_in_server_to_birthday_events` use the motor library
+for mongodb to save this information to Mongo DB Atlas.
+
+A server member may user \register-birthday to add their birthday
+to the mongo database. They may select real birthday or un-birthday,
+as well as a month and year (integers). The function `register_birthday`
+implements this and saves information to mongo.
+
+The function `create_birthday_events` exists as a task with an
+interval trigger of `interval_seconds` which is 15 for DEV_MODE
+or 25200 (7 hrs) for standard mode. This function looks at each
+guild and checks if it is opted in. Then for each guild it iterates
+over birthday_document records in descending order of creation, skipping
+documents for a user that has already been seen.
+If the event date is soon but not too soon, and the most recent event
+is more than a year ago, a new server event is scheduled and
+the last_event_datetime value is update in mongodb, as well as
+an "event_create_success" value.
+
+Function `on_startup` activates the `create_birthday_events` function.
+
+Function schedule_discord_event parses information from a provided guild,
+a mongo birthday document, and mongo opt_in_document for the guild.
+From this it calls "guild.create_scheduled_event" to create the event.
+
+Function setup adds the BirthdayEvents extension to the bot.
+"""
+import datetime
+import logging
+from http.client import HTTPException
+
+import config
+import interactions.models
+import pymongo
+from core.base import CustomClient
+from interactions import Button
+from interactions import ButtonStyle
+from interactions import Extension
+from interactions import IntervalTrigger
+from interactions import listen
+from interactions import OptionType
+from interactions import Permissions
+from interactions import slash_command
+from interactions import slash_default_member_permission
+from interactions import slash_option
+from interactions import SlashCommandChoice
+from interactions import SlashContext
+from interactions import Task
+
+# TODO: Refactor into more functions
+
+DEV_MODE = config.DEV_MODE
+
+
+class BirthdayEvents(Extension):
+    bot: CustomClient
+
+    @slash_command(name="opt-out-from-birthday-events")
```

Comment:
does discord.py allow you to add spaces to ur commands? you can do `/birthday remove (params) ` or `/birthday add (params)` instead of this very odd name lmao

Reply:
lol. I believe Discord has something similar to what you're describing in sub-commands. I haven't used them before, but I'll look into it.